<a href="https://colab.research.google.com/github/Jatingpt/Only-for-practicing/blob/main/Practice%2BExercise%2B1_(1).ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [20]:
for i in range(1,11):
  print(i*i)

1
4
9
16
25
36
49
64
81
100


In [2]:
st = "ris olleh"
rev_str = ""
for i in st:
  rev_str = i + rev_str
print(rev_str)

hello sir


In [5]:
st = "ris olleh"
rev = st[::-1]
print(rev)

hello sir


In [6]:
palin = input("Enter anything to find the palindrome: ")
if palin == palin[::-1]:
  print("It is a palindrome.")
else:
  print("It is not a palindrome.")

Enter anything to find the palindrome: 454
It is a palindrome.


In [7]:
lis = [2,3,32,2,33,2,34,3444,22,2,3322,3,433,34,32]
lis = sorted(lis)
lis[-2]

3322

In [13]:
n = int(input("Enter the number to find the whether the number is even or odd: "))
if n % 2 ==0:
  print("It is an even number.")
else:
  print("It is an odd number.")

Enter the number to find the whether the number is even or odd: 4
It is an even number.


In [19]:
a = 34
b = 43
a,b = b,a
print(a)
print(b)

43
34


In [18]:
def check_prime(n):
  if n < 1:
    print("It is not a prime number.")
  if n == 2:
    print("It is not a prime number.")
  else:
    for i in range(2,n):
      if n % i == 0:
        print("It is not a prime number.")
        break
    else:
      print("Finally, got the prime number.")
check_prime(17)

Finally, got the prime number.


In [12]:
def fibonacci(n):
  a,b = 0,1
  for i in range(n):
    print(a, end = " ")
    c = a + b
    a = b
    b = c
fibonacci(10)

0 1 1 2 3 5 8 13 21 34 

In [11]:
def fibonacci(n):
  a,b = 0,1
  for i in range(n):
    print(a)
    c = a + b
    a = b
    b = c
print(fibonacci(10))

0
1
1
2
3
5
8
13
21
34
None


In [21]:
a = [3,22,3,2,3,4,24,2,4,2,42,2]
a = set(a)
a

{2, 3, 4, 22, 24, 42}

In [22]:
a = [3,22,3,2,3,4,24,2,4,2,42,2]
unique = list(dict.fromkeys(a))
unique

[3, 22, 2, 4, 24, 42]

In [28]:
sum = 0
for i in range(1,41):
  if i % 2 == 0:
    sum += i
print(sum)

420


In [41]:
name = "Jatin"
salary = 3444
designation = "AI ML Engineer"
print(f"my name is {name}, salary is {salary}, designation is {designation}")
print("My name is {}, salary is {}, designations is {}".format(name, salary, designation))
print("My name is {0}, salary is {1}, designation is {2}".format(name, salary, designation))

my name is Jatin, salary is 3444, designation is AI ML Engineer
My name is Jatin, salary is 3444, designations is AI ML Engineer
My name is Jatin, salary is 3444, designation is AI ML Engineer


In [43]:
sum = 0
n = 1
while n <= 10:
  sum += n
  n += 1
print(sum)

55


In [50]:
#to find the first non-repeating character.
def non_repeating(n):
  for i in n:
    if n.count(i) == 1:
      return i
  return none
n = "heeeeeeeeeeeeeeeeeeehi"
non_repeating(n)

'i'

In [52]:
lis = [[1,2], [3,4], [5,6]]
flat_list = lis[0] + lis[1] + lis[2]
flat_list

[1, 2, 3, 4, 5, 6]

In [54]:
lis = [[1,2], [3,4], [5,6]]
flatten_list = [i for i in lis for i in i]
flatten_list

[1, 2, 3, 4, 5, 6]

In [57]:
a ="banana"
from collections import Counter
freq = Counter(a)
freq

Counter({'b': 1, 'a': 3, 'n': 2})

In [59]:
def rot_lis(lis, k):
  k = k % len(lis)
  return lis[-k:] + lis[:-k]
lis = [1,2,3,4,5]
k = 1
rot_lis(lis,k)

[5, 1, 2, 3, 4]

In [29]:
let = input("Enter the letter to find the vowel: ")
if let in "aeiouAEIOU":
  print("It is a vowel.")
else:
  print("Not a vowel.")

Enter the letter to find the vowel: r
Not a vowel.


In [ ]:
lis = [[1,2], [3,4], [5,6]]
flat_lis = lis[0] + lis[1] + lis[2]
flat_lis

[1, 2, 3, 4, 5, 6]

In [ ]:
lis = [[1,2], [3,4], [5,6]]
flatten_list = [i for i in lis for i in i]
flatten_list

[1, 2, 3, 4, 5, 6]

In [ ]:
st = "abc"
rev_str = st[::-1]
rev_str

'cba'

In [ ]:
def ana(str1, str2):
  if sorted(str1) == sorted(str2):
    print(str1, "and", str2, "are anagrams")
  else:
    print("Not an anagrams.")
ana("listen", "silent")

listen and silent are anagrams


In [ ]:
a = "banana"
from collections import Counter
freq = Counter(a)
freq

Counter({'b': 1, 'a': 3, 'n': 2})

In [ ]:
a = {"a": 1, "b" : 3, "c": 2}
sort_dic = sorted(a.values())
sort_dic

[1, 2, 3]

In [ ]:
a = [1,2,3]
b = [4,5,6]
import heapq
mer_lis = list(heapq.merge(a,b))
mer_lis

[1, 2, 3, 4, 5, 6]

In [ ]:

def rotat_lis(lis, k):
  k = k%len(lis)
  return lis[-k:] + lis[:-k]
l = [1,2,3,4,5,6]
k = 1
rotat_lis(l,k)

[6, 1, 2, 3, 4, 5]

In [ ]:
data = {
    "name": "John Doe",
    "age": 30,
    "email": None,
    "phone": "9876543210",
    "address": None,
    "department": "AI/ML",
    "salary": 75000,
    "joining_date": None
}

count = 0
for i in data:
  if data[i] == None:
    count += 1
print(count)

3


In [ ]:
mul = 1
a = [1,2,3]
for i in a:
  mul *= i
print(mul)


6


In [ ]:
a = "ootd.bts.grwm"
a = a.replace(".", ",")
a

'ootd,bts,grwm'

In [ ]:
a = {"a": 1, "c": 3, "b": 3}
sort_dic = sorted(a.keys())
sort_dic

['a', 'b', 'c']

In [ ]:
a = "adkfkdsfdasdf"
a = sorted(a)
a

['a', 'a', 'd', 'd', 'd', 'd', 'f', 'f', 'f', 'k', 'k', 's', 's']

In [ ]:
a = "F.R.I.E.N.D.S"
a.replace(".", "")

'FRIENDS'

In [ ]:
a = "elloe"
count = 0
for i in a:
  if i in "aeiouAEIOU":
      count += 1
print(count)

3


In [ ]:
for i in range(1,11):
  print(i*i)

1
4
9
16
25
36
49
64
81
100


In [ ]:
a = [1,2,3,22,3,4,4,2,4]
a = set(a)
a

{1, 2, 3, 4, 22}

In [ ]:
a = [1,2,3,22,3,4,4,2,4]
unique = list(dict.fromkeys(a))
unique

[1, 2, 3, 22, 4]

In [ ]:
sum = 0
for i in range(1,21):
  sum += i
print(sum)

210


In [ ]:
sum = 0
for i in range(1,41):
  if i % 2 == 0:
    sum += i
print(sum)

420


In [ ]:
a = input("Enter the letters to find the vowels: ")
if a in "aeiouAEIOU":
  print("Yes, it is a vowel.")
else:
  print("Not a vowel")

Enter the letters to find the vowels: r
Not a vowel


In [ ]:
a = int(input("Enter the number here: "))
if a % 2 == 0:
  print("It is an even number.")
else:
  print("It is an odd number.")

Enter the number here: 4
It is an even number.


In [ ]:
#press 1 to find the area of the square
#press 2 to find the area of the rectangle
find_area = int(input("Enter the number to find the area: "))
if find_area == 1:
  side = float(input("Enter the side of the square: "))
  sq_area = side * side
  print(sq_area)
elif find_area == 2:
  length = float(input("Enter the length of the rectangle: "))
  breadth = float(input("Enter the breadth of the rectangle: "))
  area_rect = length * breadth
  print(area_rect)
else:
  print("INVALID INPUT!!")

Enter the number to find the area: 2
Enter the length of the rectangle: 2
Enter the breadth of the rectangle: 3
6.0


In [ ]:
sum = 0
n = 0
while n <= 10:
  sum+=n
  n+= 1
print(sum)

55


In [ ]:
for i in range(1,20):
  if i % 2 != 0:
    print(i, end = " ")

1 3 5 7 9 11 13 15 17 19 

In [ ]:
sum = 0
n = 0
while n <= 20:
  if n % 2 != 0:
    sum += n
  n += 1
print(sum)

100


In [ ]:
def check_prime(n):
  if n <= 1:
    print("Not a prime number.")
  if n == 2:
    print("It is a prime number.")
  else:
    for i in range(2,n):
      if n % i == 0:
        print("It is not a prime number.")
        break
      else:
        print("prime number.")

In [ ]:
#non-repeating character.
def non_repeating(a):
  for char in a:
    if a.count(char) == 1:
      return char
  return None
a = "aabbcccf"
non_repeating_char = non_repeating(a)
non_repeating_char

'f'

In [ ]:
num = int(input("Enter the number to check prime: "))
if num < 1:
  print("Not a prime number.")
if num == 2:
  print("It is a prime number.")
else:
  for i in range(2,num):
    if num % i == 0:
      print("Not a prime number.")
      break
  else:
    print(num, "is a Prime number.")

Enter the number to check prime: 23
23 is a Prime number.


In [ ]:
a = "ris olleh"
rev_str = ""
for i in a:
  rev_str = i + rev_str
print(rev_str)

hello sir


In [ ]:
a = "mdam"
if a == a[::-1]:
  print("It is a palindrome.")
else:
  print("Not a palindrome.")

Not a palindrome.


In [ ]:
a = [2,3,2,4,2,3,2,4,2,4,2,4,44,32,22,45,66,4,3,2224,43]
a = sorted(a)
a[-2]

66

In [ ]:
def fact(n):
  if n == 1:
    return n
  else:
    return n * fact(n-1)
fact(5)

120

In [ ]:
def fibonacci(n):
  a,b = 0,1
  for i in range(n):
    print(a, end = " ")
    c = a + b
    a = b
    b = c
fibonacci(10)

0 1 1 2 3 5 8 13 21 34 

In [ ]:
a = 45
b = 34
a,b = b,a
print(a)
print(b)

34
45


In [ ]:
a = int(input("Write a program to check the even number: "))
if a % 2 == 0:
  print(a, "It is a even number.")
else:
  print(a, "is the odd number.")

Write a program to check the even number: 5
5 is the odd number.


In [ ]:
#Consider the string given below and answer the following questions
string = "OQYWFClFhFGAvIWYwGKpmZhnJiyzTslSIhSwvOsqJMEphzmifTkyqOMNpnOtXZxmCfgDYqbaBHAUvIWhMnvwZnEMVDvmEfLrDoQnAZgQEgXQVnmSYkfedpAdhrtpOgORpYLRZYGWdhWYuqQssCUXtTzKRDAhpjUheOzUroZNzWFtZOVwIapzUYtbSbjYNErzQ"
len(string)

193

####  Calculate the number of characters in the string

-  193
-  200
-  300
-  500


In [ ]:
# Type your code here
string.count("a")

2

#### Count the number of occurences of 'a' in the string

-  1
-  2
-  10
-  5

In [ ]:
# Type your code here

#### Identify whether the string starts with 'if'.

-  True
-  False

In [ ]:
# Type your code here
string[:2]

'OQ'

#### Return the substring starting at index 63 and ending at index 88  [Both the indexes are inclusive]

-  'OQYWFClFhFGA'
-  'mCfgDYqbaBHAUvIWhMnvwZnEMV'
-  'IWYwGKpmZhnJiyzTslSIhSwv'
-  'JMEphzmifTkyqOMNpnOtXZ'

In [ ]:
# Type your code here
string[63:89]

'mCfgDYqbaBHAUvIWhMnvwZnEMV'

#### Identify the character at index 45

-  a
-  b
-  z
-  x

In [ ]:
# Type your code here
string[45]

'z'

In [ ]:
#WSCube
#Progam to display name and others.
#swap two variables
#convert float into int
#take input from user and convert it into int

In [ ]:
name = "jatin"
age = 24
print(name)
print(age)

print()

a = 34
b = 33
a,b = b,a
print(a)
print(b)

print()

a = 1.23
a = int(a)
print(a)

print()

n = float(input("Enter the float value: "))
print(n)
n = int(n)
print(n)

jatin
24

33
34

1

Enter the float value: 4
4.0
4


In [ ]:
#identity operator
a = 123
b = "345"
print(a is b)

#membership operator
a = 'abcde'
print("s" in a)

False
False


In [ ]:
#write a program to check the number is positive.
#to check odd or even number
#create area calculator
#to check whether the passed letter is vowel or not
#to check if the passed number is single digit or double digit.

In [ ]:
def check(n):
  if n >= 1:
    print("Positive")
  else:
    print("Negative")
check(-3)

def odd_even(n):
  if n % 2 == 0:
    print("It is a even number.")
  else:
    print("odd number")
odd_even(5)

                                         #Area Calculator
#press 1 to get the area of the square.
#press 2 to get the area of the rectangle

# area = int(input("Enter the input to find the area: "))
# if area == 1:
#   side = float(input("Enter the side of the square to get the area: "))
#   area_square = side * side
#   print(area_square)
# elif area == 2:
#   length = float(input("Enter the side of the rectangle: "))
#   breadth = float(input("Enter the side of the rectangle: "))
#   area_rect = length * breadth
#   print(area_rect)
# else:
#   print("INVALID INPUT, Kindly put the valid input to find the area.")


# vow = input("Enter the letter to find the vowel")
# if vow in 'AEIOUaeiou':
#   print("The passed letter is a vowel")
# else:
#   print("Not a vowel")



Negative
odd number


In [ ]:
#print the table of 5 using for and while loop each.
#write a program for nested loop
#use continue statement in for loop and while loop
#find the sum of all even number upto 50
#write first 10 number and their squares.
#check how many numbers are divided by 8 and 12 both

In [ ]:
# n = 5
# for i in range(1,11):
#   print(n*i)

# n = 1
# a = 5                        #By the input method can canculate any table:
# while n <=10:
#   print(a * n)
#   n += 1

for i in range(1,5):
  if i == 3:
    print("skip this part please.")
    continue
  else:
    print(i)

for i in range(1,5):
  if i == 3:
    print("Song is ended")
    break
  else:
    print(i)

sum = 0
for i in range(1,51):
  if i % 2 == 0:
    sum += i
print(sum)

for i in range(1,11):
  print("The square of", i, "is", i*i)


for i in range(1,101):
  if i % 8 == 0 and i % 12 == 0:
    print(i)


1
2
skip this part please.
4
1
2
Song is ended
650
The square of 1 is 1
The square of 2 is 4
The square of 3 is 9
The square of 4 is 16
The square of 5 is 25
The square of 6 is 36
The square of 7 is 49
The square of 8 is 64
The square of 9 is 81
The square of 10 is 100
24
48
72
96


In [ ]:
#problems on string

a = "why to fit in, when you are born to shine."
#write the program to find the length of the strng.
#write a program to check how many times alphabet o is occuring
#write the program to convert the whole string into lower and upper
#write the program to convert this string into a single title
#write the program to find index of fit

In [ ]:
print(len(a))

print(a.count("o"))

print(a.lower())

print(a.upper())

print(a.title())

print(a.index("fit"))



42
4
why to fit in, when you are born to shine.
WHY TO FIT IN, WHEN YOU ARE BORN TO SHINE.
Why To Fit In, When You Are Born To Shine.
7


In [ ]:
#fibonacci series
#palindrome
#check the digit is prime or not

In [ ]:
def fibonacci(n):
  a,b = 0,1
  for i in range(1,11):
    print(a, end = " ")
    c = a + b
    a = b
    b = c
fibonacci(10)

print()

def palindrome(n):
  if n == n[::-1]:
    print("It is a palindrome.")
  else:
    print('Not a palindrome.')
palindrome("madam")


def check_prime(n):
  if n <= 1:
    print("It is not a prime number.")
  else:
    for i in range(2,int(n**0.5)+1):
      if n % i == 0:
        print("Not a prime number.")
        break
    else:
      print("Prime number.")
check_prime(3)

0 1 1 2 3 5 8 13 21 34 
It is a palindrome.
Prime number.


In [ ]:
a = "ootd.yolo.asap"
#write the program to separate the following string into comma separated values
#write the program to sort alphabetically in python
z = "F.R.I.E.N.D.S"
#write the program to remove a given character from a string
#remove dot(.)

In [ ]:
a = "ootd.yolo.asap"
b = a.replace(".",",")
b


s = "adgde"
s = sorted(s)
print(s)

z = "F.R.I.E.N.D.S"
b = z.replace(".", "")
print(b)

['a', 'd', 'd', 'e', 'g']
FRIENDS


In [ ]:
#take the input from the user and reverse it
#write the program to check string contains only digit.
#write the program to find the number of vowels in a string
#write the program to check if every word in a string begins with a capital letter.

In [ ]:
# inp = input("Enter anything: ")
# print(inp[::-1])


a = "345"
print(a.isdigit())

a = "hello brother, where are you."
vowel = 0
for i in a:
  if i == "a" or i == "A" or i == "e" or i == "E" or i == "i" or i == "I" or i == "o" or i == "O" or i == "u" or i == "U":
    vowel += 1
print(vowel)


a = "Hello Bhai, how are You"
b = "Hello Brother, How Are You"
print(a.istitle())
print(b.istitle())

True
10
False
True


In [ ]:
#functions of the list
a = ["hulk", "thor", "spiderman", "batman"]
len(a)
a.count("hulk")
a.append("superman")
a.insert(1,"krish")
a.remove("krish")

In [ ]:
a = ["hulk", "thor", "spiderman", "batman"]
b = ["boomerman"]
a.extend(b)
print(a)

print(a.reverse())

['hulk', 'thor', 'spiderman', 'batman', 'boomerman']
None


In [ ]:
#list comprehension
l1 = [1,2,3,4]
l2 = []
for i in l1:
  l2.append(i)
print(l2)

[1, 2, 3, 4]


In [ ]:
l1 = [30,40,50,60]
l2 = []
for i in l1:
  if l1 > 45:                   #will check later
    l2.append(i)
print(l2)

TypeError: '>' not supported between instances of 'list' and 'int'

In [ ]:
#write a program to swap first and fourth element.

a = ["Ross", "Rachel", "Monica", "Joe"]
a[1],a[3] = a[3],a[1]
print(a)

#write a program to add a new element at second position
a.insert(1,"Phoebe")
print(a)

#write a program to delete a values from 3rd position
a = [1,2,3,4,5,6]
a.remove(a[2])
print(a)

#write a program to multiply all the numbers in a list
a = [1,2,3,4,5,6]
mul = 1
for i in a:
  mul *= i
print(mul)


#write a program to get the largest number from a list
max(a)
min(a)

['Ross', 'Joe', 'Monica', 'Rachel']
['Ross', 'Phoebe', 'Joe', 'Monica', 'Rachel']
[1, 2, 4, 5, 6]
720


1

In [ ]:
##########                Tuple Questions

a = ("Oneplus", "Nokia", "Motorola", "Vivo")
for i in a:
  print(i,end = " ")

print()

i = 0
while i < len(a):
  print(a[i])
  i += 1

Oneplus Nokia Motorola Vivo 
Oneplus
Nokia
Motorola
Vivo


In [ ]:
stu_data = {"name": "David", "age": 34, "marks": 65}

#write a program to get the student data.
stu_data["age"]


##### functions of dictionary

print(stu_data.keys())
print(stu_data.values())
print(stu_data.get("name"))
print(stu_data["name"])


dict_keys(['name', 'age', 'marks'])
dict_values(['David', 34, 65])
David
David


In [ ]:
#Creating a nested dictionary
emp = {1: {"name": "Saroj", "age": 34},
       2: {"name": "kritesh", "age": 59}}
print(emp)

#how can i get age of the saroj
emp[1]["age"]
emp[2]["name"]

{1: {'name': 'Saroj', 'age': 34}, 2: {'name': 'kritesh', 'age': 59}}


'kritesh'

In [ ]:
stu_data = {"rank": 89, "age": 34, "marks": 65}

#write a program to set a dictionary by values
stu_data = sorted(stu_data.values())
print(stu_data)

#write a program to print a dictionary where the keys are the number between 1 and 10 and their values are their squares.
mul = {}
for i in range(1,11):
  mul[i] = i**2
print(mul)

[34, 65, 89]
{1: 1, 2: 4, 3: 9, 4: 16, 5: 25, 6: 36, 7: 49, 8: 64, 9: 81, 10: 100}


In [ ]:
#write a program to multiply all the items in a dictionary.
a = {"a": 1, "b": 2, "c": 3, "d": 4}
mul = 1
for i in a:
  mul *= a[i]
print(mul)


24


In [ ]:
#write a program to sort a dictionary by key
a = {"g": 1, "b": 2, "c": 3, "d": 4}
a = sorted(a.keys())
print(a)

['b', 'c', 'd', 'g']


In [ ]:
#######                    Sets functions
a = {"Ironman", "Hulk", "Spiderman"}
a.add("Batman")
print(a)

a.pop()
print(a)

a.remove("Batman")
print(a)

a.discard("Spiderman")
print(a)

{'Hulk', 'Ironman', 'Batman', 'Spiderman'}
{'Ironman', 'Batman', 'Spiderman'}
{'Ironman', 'Spiderman'}
{'Ironman'}


In [ ]:
a = {1,2,3,4,2,23232,2,32,4,54,3}


##write a program to find the max and minimum values in a set
max(a)
min(a)

a = {1,2,3}
b = {3,4,5}
c = {3,6,4}


##write a program to find common elements in three sets
d = a.intersection(b)
e = d.intersection(c)

print(e)


a = {1,2,3}
b = {3,4,5}

#write a program to find the difference between two sets
print(a.difference(b))


{3}
{1, 2}


In [ ]:
###############  Functions

#arbitary function

def hello(*name):
  print("My name is", name[2])
hello("Jotshna", "rakesh", "punk")

My name is punk


In [ ]:
a = lambda b: b*5
a(4)

x = lambda a,b,c: (a+b)*c
print(x(1,2,3))

9


In [ ]:
#write a function to find the maximum of three numbers

def max_num(a,b,c):
  if a > b and a > c:
    print("Max number is", a)
  elif b > a and b > c:
    print("Max number is", b)
  else:
    print("Max number is", c)
max_num(3,4,2)

Max number is 4


In [ ]:
#write a function to create and print a list where the values are square of numbers between 1 and 15.
def creat_list():
  l = []
  for i in range(1,16):
    l.append(i**2)
  return l
print(creat_list)


<function creat_list at 0x78a207801580>


In [ ]:
#write a function that takes no as a parameter and check if the number is prime or not.
n = int(input("Enter the number to check prime or not: "))
def check_prime(n):
  if n <= 1:
    print("Not a prime number.")
  else:
    for i in range(2,int(n**0.5)+1):
      if n % i == 0:
        print("Not a prime number.")
        break
    else:
      print("Prime number.")
check_prime(n)

Enter the number to check prime or not: 8
Not a prime number.


In [ ]:
#write a function to sum all the numbers of a list.
def add_num(number):
    total = 0
    for i in number:
        total += i
    return total
add_num([1,2,3,4,5])

15

In [ ]:
#write a function for fibonacci series
def fibonacci(n):
  a,b = 0,1
  for i in range(n):
    print(a, end = " ")
    c = a + b
    a = b
    b = c
fibonacci(10)

0 1 1 2 3 5 8 13 21 34 